### PR TITLE
RSDK-2828 - Add motor example

### PR DESCRIPTION
**Pull request by clintpurser**
This PR adds an example screen for motors in the example app.

https://user-images.githubusercontent.com/22106496/236237245-12de6de0-6526-4dd4-b6e3-523c27313601.mov

Screenshot 2023-05-04 at 10 25 20 AM Screenshot 2023-05-04 at 10 10 34 AM

### CODE REVIEW

**Comment by clintpurser** [comment]
this will conflict with https://github.com/viamrobotics/viam-flutter-sdk/pull/13 - I will wait for that to get merged before merging this.

**Comment by clintpurser** [comment]
This is ready for review again.

**Comment by clintpurser** [comment]
I pulled in an extra commit with my rebase, i'll fix here shortly then unmark as draft.

**Comment by clintpurser** [comment]
Fixed my rebase issues, this is ready for review! 🥳

File: example/viam_example_app/lib/main.dart
Comment (stuqdog) [inline example/viam_example_app/lib/main.dart:136]: This looks duplicative with the code a couple lines up. Am I missing some dart-ism?

File: example/viam_example_app/lib/main.dart
Comment (stuqdog) [inline example/viam_example_app/lib/main.dart:167]: This empty line here seems a bit strange to me, is this a dart idiom?

File: example/viam_example_app/lib/screens/motor.dart
Comment (stuqdog) [inline example/viam_example_app/lib/screens/motor.dart:66]: This is really heavily nested, it makes it kinda hard for me to read/parse. I don't really know dart at all so I'm happy to defer if this is typical, but I wonder if there's a way we could factor chunks out so that this is a bit clearer of a workflow?

File: example/viam_example_app/lib/screens/motor.dart
Comment (stuqdog) [inline example/viam_example_app/lib/screens/motor.dart:37]: Recurrence of the empty line here makes me think this is just a typical dart thing, but figured wanted to flag here as well as above just in case :)

**Review by stuqdog — COMMENTED** [review]
A couple questions mostly. I don't know dart at all so please take this with a grain of salt!

File: example/viam_example_app/lib/main.dart
Comment (clintpurser) [reply example/viam_example_app/lib/main.dart:136]: Good catch, thank you! This is duplicate code that needs to be removed. will push up change momentarily... 🙏

Comment (clintpurser) [reply example/viam_example_app/lib/main.dart:128]: I decided to remove the higher, preferring alphabetization.

File: example/viam_example_app/lib/screens/motor.dart
Comment (clintpurser) [reply example/viam_example_app/lib/screens/motor.dart:66]: This is a good callout, I considered doing this multiple times but in the end decided not to for this screen. In a larger more complex screen, I agree, breaking things out into chunks would be a must. because of the simplicity of this screen and the lack of reused widgets (the visual components) I decided not factor things out into their own chunks.

Flutter is naturally very nested, so it may just need some adjusting when you're looking at a flutter view vs a dart file. Happp to take this offline and walk you (or anyone) whats going on in this file, if that would help it make more sense.

File: example/viam_example_app/lib/screens/motor.dart
Comment (clintpurser) [reply example/viam_example_app/lib/screens/motor.dart:37]: yes, similar as before.

The PlatformText widget in this case automatically formatted to take up multiple lines because of the line lengths in the separate fields, and how indented in it is in the tree. (This is a setting you can dial in in the flutter/dart linter).

Essentially what is going on here is:
```
PlatformText(String 'text', Style TextSty.(params),),
```

but it gets broken out to

```
PlatformText(
   'text',
   Style: TextStyle(params...),
),
```

if the arguments for `TextStyle(...)` were more verbose then that would potentially get broken out too.

It doesn't affect how the code gets executed, this is just an attempt to standardize line length / readability automatically.

Comment (clintpurser) [reply example/viam_example_app/lib/screens/motor.dart:37]: @stuqdog I aligned the settings between the outer project and the nested example project in this PR https://github.com/viamrobotics/viam-flutter-sdk/pull/16

File: example/viam_example_app/lib/screens/motor.dart
Comment (stuqdog) [reply example/viam_example_app/lib/screens/motor.dart:66]: No need, appreciate the offer! I think I can follow along, just a little more eye-movement than usual :)

Comment (clintpurser) [reply example/viam_example_app/lib/screens/motor.dart]: I was thinking about this feedback this morning and looked through the PR again and realized there was an extra nested column that was unneeded. I removed it, and there is one less layer of nesting now. I am glad you brought this up 😄 ✊

**Review by maximpertsov — APPROVED** [review]

**Review by stuqdog — APPROVED** [review]

**Pull request merged**
